<?xml version="1.0"?>
<launch>

    <!-- sudo chmod 777 /dev/ttyUSB0 -->

    <!-- Robot args: Gazebo vs. Real Robot Arm -->
    <arg name="sim" default="false" />
    <arg name="urdf" default="true" />

    <!-- Robot params: Gripper --> 
    <arg name="comport" value="/dev/ttyUSB0"/>
    <arg name="baud" value="115200"/>

    <!-- Description args: Joint state publisher gui -->
    <arg name="gui" default="false" />

    <!-- Start the arm driver -->
    <include file="$(find comanip_config)/launch/kinova_robot.launch" unless="$(arg sim)">
        <arg name="kinova_robotType" value="j2s7s300" />
        <arg name="use_urdf" value="$(arg urdf)" />
    </include>

    <!-- Start the trajectory server driver -->
    <node pkg="kinova_driver" type="hlpr_trajectory_controller" name="jaco_trajectory_controller" output="screen">
        <param name="sim" value="$(arg sim)"/>
    </node>

    <!-- Start the gripper driver and gripper action server driver -->
    <node pkg="robotiq_85_driver" type="robotiq_85_driver" name="robotiq_85_driver" respawn="true" output="screen">
        <param name="num_grippers" value="1"/>
        <param name="comport" value="$(arg comport)" />
        <param name="baud" value="$(arg baud)" />
    </node>
    <node pkg="robotiq_85_gripper_actions" type="gripper_actions" name="gripper_actions"/>


    <!-- Aruco tracker -->
<!--     <node pkg="aruco_tracker" name="set_camera_info" type="set_camera_info.py" output="screen"/> -->
	
    <include file="$(find comanip_config)/launch/segmentation_camera.launch"/>

    <!-- Start the description publishers -->
    <include file="$(find comanip_description)/launch/comanip_description.launch"/>

    <!--Start the moveit driver-->
    <include file="$(find comanip_moveit)/launch/move_group.launch"/>

</launch>
